DOMjudge_tag_release.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+. release.env
+
+for i in `git tag | grep 7`; do git tag -d $i; done
+
+DOWNLOAD_RELEASE_SCRIPT=""
+
+notify_channel () {
+    # Optionally notify the Slack channel
+    # that a build has been triggered
+}
+
+test_tarball () {
+    # Happens as current side effect of docker build
+}
+
+build_deb () {
+    # Not sure if this should happen on the server?
+}
+
+build_docker () {
+    # Create on: https://gitlab.com/DOMjudge/domjudge-packaging/-/settings/ci_cd
+    curl -X POST \
+         -F token=$TOKEN \
+         -F "ref=master" \
+         -F "variables[DOMJUDGE_VERSION]=$1" \
+         https://gitlab.com/api/v4/projects/24097565/trigger/pipeline
+}
+
+process_tag () {
+    if [ ! -f $RELEASE_DIR/$1 ]; then
+        notify_channel "Found new tag: ($1)."
+        if [ -z $DOWNLOAD_RELEASE_SCRIPT ]; then
+            wget $RELEASE_WEB
+            $DOWNLOAD_RELEASE_SCRIPT="done"
+        fi
+        ./make_release $1
+        rsync -rva ./$1* $RELEASE_DIR/
+        notify_channel "Tarball finished ($1)."
+        test_tarball $1
+        build_docker $1
+        build_deb $1
+        if [ $2 =eq 1 ]; then
+            # Release on the main branch,
+            # This is the latest release
+            #
+            # TODO: Handle website update
+        fi
+    fi
+}
+
+process_branch () {
+    while read tag
+    do
+        echo "Handling tag: " $tag " in branch: " $1
+        process_tag $tag
+    done <<< $(git tag)
+}
+
+# Reset to master
+git checkout master
+
+process_branch origin/master 1
+
+while read branch
+do
+    process_branch $branch 0
+done <<< $(git branch -r | grep "origin")
```

Comment:
This calling `process_branch` and then `process_tag` all seems a bit complicated and unclear to me. For example `git tag` will give all tags, not just the ones on a particular branch.

So why not a single loop only over all tags? Then also it should be checked that that tag is a release tag, because we do have other tags too. So likely it should match `[0-9]+\.[0-9]+\.[0-9]+((RC|rc)[0-9])?`.

Reply:
Somehow locally for me `git tag` does not give all the tags. Which could be that locally I don't checkout in the correct way.

But I will implement your suggestions, as it indeed is a better way for the tags (and the other thing can be spotted easy enough if you expect a build which is not happening)